rs/ic_os/config/src/lib.rs

```diff
@@ -1,17 +1,60 @@
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
 use std::collections::HashMap;
 use std::fs::read_to_string;
+use std::fs::{create_dir_all, File};
+use std::io::Read;
+use std::io::Write;
+use std::net::{Ipv4Addr, Ipv6Addr};
 use std::path::Path;
 
+use anyhow::bail;
 use anyhow::{Context, Result};
+use url::Url;
+use utils::deployment::read_deployment_file;
+
+pub mod types;
+use crate::types::NetworkSettings;
 
 pub type ConfigMap = HashMap<String, String>;
 
-pub static DEFAULT_SETUPOS_CONFIG_FILE_PATH: &str = "/var/ic/config/config.ini";
+// todo: update naming once config variables start applying to all config partitions
+pub static DEFAULT_SETUPOS_CONFIG_OBJECT_PATH: &str = "/var/ic/config/config.json";
+// pub static DEFAULT_CONFIG_OBJECT_PATH: &str = "/tmp/config/config.json";
+
+pub static DEFAULT_SETUPOS_CONFIG_FILE_PATH: &str = "/config/config.ini";
 pub static DEFAULT_SETUPOS_DEPLOYMENT_JSON_PATH: &str = "/data/deployment.json";
+pub static DEFAULT_SETUPOS_NNS_PUBLIC_KEY_PATH: &str = "/data/nns_public_key.pem";
+pub static DEFAULT_SETUPOS_SSH_AUTHORIZED_KEYS_PATH: &str = "/config/ssh_authorized_keys";
+pub static DEFAULT_SETUPOS_NODE_OPERATOR_PRIVATE_KEY_PATH: &str =
+    "/config/node_operator_private_key.pem";
 
+//todo: delete
 pub static DEFAULT_HOSTOS_CONFIG_FILE_PATH: &str = "/boot/config/config.ini";
 pub static DEFAULT_HOSTOS_DEPLOYMENT_JSON_PATH: &str = "/boot/config/deployment.json";
 
+pub fn get_deployment_settings(
+    deployment_json_path: &Path,
+) -> (u32, String, Vec<Url>, String, String) {
+    match read_deployment_file(deployment_json_path) {
+        Ok(deployment_json) => (
+            deployment_json.resources.memory,
+            deployment_json
+                .resources
+                .cpu
+                .clone()
+                .unwrap_or("kvm".to_string()),
+            deployment_json.nns.url.clone(),
+            deployment_json.deployment.name.to_string(),
+            deployment_json.logging.hosts.to_string(),
+        ),
+        Err(err) => {
+            eprintln!("Error retrieving deployment file: {err}. Using default values.");
+            default_deployment_values()
```

Comment:
See default_deployment_values(). This will save our butts in the case where deployment.json can't be parsed (like the last HostOS upgrade issue). And these deployment.json values should be untouched on mainnet, correct?